modules/ocf/manifests/firewall/allow_mosh.pp

```diff
@@ -0,0 +1,12 @@
+# firewall input rule to allow mosh
+class ocf::firewall::allow_mosh {
```

Comment:
I'm not sure if this is the right abstraction, since it doesn't make sense to include `ocf::firewall::allow_mosh` without also including `ocf::firewall::allow_ssh`, as mosh creates the initial connection using ssh. I would instead create an `ocf::firewall::allow_ssh_and_mosh` class that includes `ocf::firewall::allow_ssh` and defines the additional rules for mosh. Then `ocf_ssh`, `ocf_apphost`, `ocf_admin`, and `ocf_kvm` can include `ocf::firewall::allow_ssh_and_mosh` instead.

Reply:
Or we could just move `allow_mosh` into `allow_ssh` without using a new name.

If we for some reason want to allow ssh without allowing mosh, we should simply add `require allow_ssh` to `allow_mosh`.

Reply:
Drawback of that though is it's not obvious from the name that `allow_mosh` also allows ssh

Reply:
I think it will be obvious to someone basically informed of what mosh is, which someone touching this code obviously should be.

Reply:
No, I meant the forward proposition.

I on the contrary don't think it's a given that anyone reading these rules will already know what mosh is. They might be confused that hosts with the rule are allowing SSH, even though there's apparently no `allow_ssh` rule.

Having said that, we are getting into the bikeshed zone...

Reply:
> They might be confused that hosts with the rule are allowing SSH, even though there's apparently no allow_ssh rule.

That's possible, although I was envisioning a scenario in which one class would `include allow_ssh` and another class would include it and add `include allow_mosh` to that. How about we just merge `allow_mosh` into `allow_ssh` and relieve ourselves of this question completely?